src/client.rs

```diff
@@ -133,15 +133,9 @@ impl DatagoClient {
 
         if let Some(engine) = &mut self.engine {
             // let _ = engine.samples_meta_rx.close();
-            let _ = engine.pages_rx.close();
+            let _ = engine.samples_metadata_rx.close();
```

Comment:
this alone will passively close the feeder thread
l137 will passively close the worker thread

-> we can join them just after

Reply:
previously the dispatch thread was implicitly stopping because there was no more incoming data, now is a little more explicit (every thread gets a signal), probably cleaner